src/model/wallet.ts

```diff
@@ -491,9 +531,19 @@ export const oldLoadSeedPhrase = async (): Promise<null | EthereumWalletSeed> =>
 export const loadAddress = (): Promise<null | EthereumAddress> =>
   keychain.loadString(addressKey) as Promise<string | null>;
 
-const loadPrivateKey = async (
+export const loadPrivateKey = async (
   address?: EthereumAddress | undefined
 ): Promise<null | EthereumPrivateKey | -1 | -2> => {
+  const addressToUse = address || (await loadAddress());
+  if (!addressToUse) {
+    return null;
+  }
```

Comment:
this function is extremely delicate my suggestion is you leave it as it is and move this logic to whoever is calling it.
If something goes wrong you'll end up grabbing the wrong pkey